x-pack/metricbeat/module/sql/query/query.go

```diff
@@ -142,9 +160,24 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
 				return fmt.Errorf("fetch variable mode failed: %w", err)
 			}
 
-			m.reportEvent(ms, reporter, q.Query)
+			if m.Config.MergeResults {
+				for k, v := range ms {
+					_, ok := merged[k]
+					if ok {
+						m.Logger().Warn("overwriting duplicate metrics: ", k)
+					}
+					merged[k] = v
+				}
+			} else {
+				// Report immediately for non-merged cases.
+				m.reportEvent(ms, reporter, q.Query)
+			}
```

Comment:
Nit. This is getting more complicated now, maybe the code for variable and table format could be moved to separated functions.